tests/scripts/thread-cert/v1_2_test_csl_transmission.py

```diff
@@ -71,6 +73,26 @@ def test(self):
         self.assertTrue(self.nodes[LEADER].ping(self.nodes[SSED_1].get_rloc()))
         self.simulator.go(5)
 
+        ssed_messages = self.simulator.get_messages_sent_by(SSED_1)
+        msg = ssed_messages.next_mle_message(mle.CommandType.CHILD_UPDATE_REQUEST)
+        msg.assertMleMessageContainsTlv(mle.CslChannel)
+
+        # Change channel
+        self.nodes[LEADER].send_mgmt_pending_set(channel=SECOND_CHANNEL)
```

Comment:
For non-commissioner thread device, it should include full pending dataset in MGMT_PENDING_SET, otherwise, the pending dataset in MLE Data Response would be incomplete.
To allow partial dataset from simplification's perspective, I would suggest you to make Leader become active commissioner.